Dockerfile

```diff
@@ -5,10 +5,15 @@
 # under the terms of the MIT License; see LICENSE file for more details.
 
 # Install base image and its dependencies
-FROM python:3.8-slim-buster
-# hadolint ignore=DL3008, DL3013, DL3015
+FROM ubuntu:20.04
+
+ENV DEBIAN_FRONTEND=noninteractive
```

Comment:
Cosmetics: since we are cleaning the `Dockerfile` that could serve other repositories, what about adding the comments to each paragraph and cleaning their text?  E.g. we don't "install" base image, we just use it...

Something like:

```
# Use Ubuntu LTS base image
FROM ubuntu:20.04

# Use default answers in installation commands
ENV DEBIAN_FRONTEND=noninteractive

# Prepare list of Python dependencies
COPY requirements.txt /code/

# Install all system and Python dependencies in one go
# hadolint ignore=DL3008, DL3013
RUN apt-get update  -y && \
    apt-get install --no-install-recommends -y \
      autoconf \
      ...

# Copy cluster component source code
WORKDIR /code
COPY . /code

# Are we debugging?
ARG DEBUG=0
RUN if [ "${DEBUG}" -gt 0 ]; then pip install -e ".[debug]"; else pip install .; fi;

# Are we building with locally-checked-out shared modules?
# hadolint ignore=SC2102
RUN if test -e modules/reana-commons; then pip install -e modules/reana-commons[kubernetes] --upgrade; fi

# Check for any broken Python dependencies
RUN pip check
```

It could later serve as a model for `Dockerfile` in other repositories.